src/styles/modal.css

```diff
@@ -0,0 +1,86 @@
+.modal {
+  position: fixed;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  background: rgba(24, 31, 36, 0.7);
+  z-index: 999999;
+}
+
+.modal-main {
+  position: fixed;
+  background: #2d1769;
+  min-width: 40vw;
+  max-width: 70vw;
+  height: max-content;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  border-radius: 12px;
+  padding: 30px;
+  box-shadow: 0px 4px 10px rgba(0,0,0,0.5);
+}
+
+.closeModal-icon {
+  position: absolute;
+  top: 13px;
+  right: 13px;
+  height: 20px;
+  width: 20px;
+  background-color: transparent;
+  border: none;
+  cursor: pointer;
+}
+
+.modal-main-title {
+}
+.modal-main-content {
+  min-width: 70%;
+  margin-bottom: 40px;
+  max-width: 100%;
+  text-align: center;
+}
+
+.modal-main-buttons {
+  justify-self: flex-end;
+}
+
+.button-modal {
+  text-transform: uppercase;
+  font-weight: 700;
+  letter-spacing: 1.5px;
+  width: 150px;
+}
+
+.button-close {
+  background-color: transparent;
+  border: 2px solid white;
+  margin-right: 20px;
+}
+
+.button-action {
+  background-color: var(--main-purple);
+}
+
+.display-block {
+  display: block;
+}
+
+.display-none {
+  display: none;
+}
```

Comment:
instead of this, I would just add the class "modal" and set the style to `display: block` and then have another class "hidden" that is applied when `!shown` and that over-rides the style with `display: none`, or vice-versa.